concordium-consensus/src/Concordium/KonsensusV1/Scheduler.hs

```diff
@@ -363,7 +370,13 @@ processPaydayRewards (Just PaydayParameters{..}) theState0 = do
     -- in which the rewards are distributed.
     foundationAddr <- getAccountCanonicalAddress =<< bsoGetFoundationAccount theState0
     theState1 <- doMintingP6 paydayMintRate foundationAddr theState0
-    distributeRewards foundationAddr paydayCapitalDistribution paydayBakers paydayPoolRewards theState1
+    theState2 <- distributeRewards foundationAddr paydayCapitalDistribution paydayBakers paydayPoolRewards theState1
+    -- We prime all active validators of the payday for possible suspension the next snapshot.
+    case hasValidatorSuspension of
+        STrue -> bsoPrimeForSuspension theState2 (Map.keys paydayPoolRewards)
+        SFalse -> return theState2
```

Comment:
Don't we only want to prime the validators that are over the threshold for missed blocks? Also, I would suggest using `paydayBakers` as the source of the list of `BakerID`s rather than `paydayPoolRewards` (e.g. `paydayBakers ^.. each . bakerIdentity`).

Reply:
I think we should also consider generating a special outcome when a validator is primed to help with notifying inactive validators that they face suspension.

Reply:
Added the special outcome for the validator when it gets primed.